cmd/kafka-example/main.go

```diff
@@ -57,40 +65,82 @@ func main() {
 		close(ch)
 	}(stdinChannel)
 
-	shutdownGracefully := func() {
+	done := make(chan bool)
+	go func() {
```

Comment:
IMO i think it does, if the go routine is removed it will unnecessarily block from ever reaching the next bit of code (the graceful shutdown), unless i wrapped all of that in it's own go routine and moved it above for loop.

I can still move the signal channel listening in this select, but must then call close(eventLoopDone).

Reply:
as discussed:  the goroutine is fine, but the signal handling needs to be outside of it, now